### PR TITLE
Sorting and filtering fixes

### LINKED ISSUE #179: SOB bounty sort/filter option
@arash009 commented on [Tue Nov 28 2017](https://github.com/status-im/ideas/issues/48)

## Preamble

    Idea: #48
    Title: SOB Bounty sort/filter option
    Status: Draft
    Created: 2017-11-28

## Summary
There is no way currently to sort or filter the list of bounties on the page. A sort and filter option would be very useful as our list of active bounties start to add up.

## Vision
As a developer looking for bounties to work on, I would like to be able to search, filter, and sort the bounties available, in order that I can find suitable bounties that I'm interested in without having to scroll through a large list.

## Swarm Participants

- Lead Contributor: 
- Testing & Evaluation: 
- Contributor: 
- Contributor: 
- UX (if relevant): @denis-sharypin

## Requirements
Sort options should include:
- Date
- Bounty value
- Repo/Org

Filter options should include:
- Repo/Org
- Min/Max bounty value
- Bounty type? (bug/feature/security/non-development)
- Programming language
- Date

### Minimum Viable Product

Goal Date:  
Description: 

### Iteration 1..N
Goal Date:  
Description: 

## Copyright
Copyright and related rights waived via [CC0](https://creativecommons.org/publicdomain/zero/1.0/).

### PR DESCRIPTION
**Pull request by pablodip**
Fixes #179 

Status: Finished

### CODE REVIEW

File: src/cljs/commiteth/ui_model.cljs [hunk 1]
```diff
@@ -9,7 +9,7 @@
 (def bounty-sorting-types-def
   {::bounty-sorting-type|most-recent   {::bounty-sorting-type.name               "Most recent"
                                         ::bounty-sorting-type.sort-key-fn        (fn [bounty]
-                                                                                   (:created-at bounty))
+                                                                                   (:updated-at bounty))
```
Comment (siphiuel) [inline src/cljs/commiteth/ui_model.cljs]: Please change to `:updated`.

File: src/cljs/commiteth/ui_model.cljs [hunk 2]
```diff
@@ -9,7 +9,7 @@
...
                                         ::bounty-sorting-type.sort-comparator-fn compare}
```
Comment (siphiuel) [inline src/cljs/commiteth/ui_model.cljs]: I think sort order should be reversed here, e.g. `(comp - compare)`

File: src/cljs/commiteth/ui_model.cljs [hunk 3]
```diff
@@ -59,11 +59,11 @@
       (t/interval filter-from (t/now)))))
 (def bounty-filter-type-date-predicate
   (fn [filter-value-interval bounty]
-    (when-let [created-at-inst (:created-at bounty)]
-      (let [created-at-date (-> created-at-inst inst-ms t-coerce/from-long)]
-        (t/within? filter-value-interval created-at-date)))))
+    (when-let [date-inst (:updated-at bounty)]
```
Comment (siphiuel) [inline src/cljs/commiteth/ui_model.cljs]: `:updated-at` -> `:updated`

**Comment by pablodip** [comment]
@siphiuel Changes made, good catches! :)

**Pull request merged**